### PR TITLE
feat: Replace `alertMixin` usage with `useAlert`

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR will replace the usage of `alertMixin` from the code base with the `useAlert` composable.

Fixes https://linear.app/chatwoot/issue/CW-3462/replace-alertmixin-usage-with-usealert

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## How Has This Been Tested?

Please refer this issue description
https://linear.app/chatwoot/issue/CW-3462/replace-alertmixin-usage-with-usealert

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/components/SnackbarContainer.vue [hunk 1]
```diff
@@ -42,7 +40,7 @@ export default {
     this.$emitter.off('newToastMessage', this.onNewToastMessage);
   },
   methods: {
-    onNewToastMessage(message, action) {
+    onNewToastMessage({ message, action }) {
```
Comment (scmmishra) [inline app/javascript/dashboard/components/SnackbarContainer.vue:43]: Why the need to convert this to an object?

Comment (scmmishra) [reply app/javascript/dashboard/components/SnackbarContainer.vue:43]: Oh just noticed the change in `composables/index.js` Any reason to do that?

File: app/javascript/dashboard/components/app/PaymentPendingBanner.vue
Comment (scmmishra) [inline app/javascript/dashboard/components/app/PaymentPendingBanner.vue:1]: Let's do this in a separate PR

**Review by scmmishra — APPROVED** [review]

File: app/javascript/dashboard/components/SnackbarContainer.vue (hunk 1 above)
Comment (scmmishra) [reply app/javascript/dashboard/components/SnackbarContainer.vue:43]: Fixes a bug from mitt migration

Comment (iamsivin) [reply app/javascript/dashboard/components/SnackbarContainer.vue:43]: @scmmishra The reason is that there is a component where we pass the action to the `useAlert`. So when I pass it as two params, the action does not show up, so I do it as an object. https://github.com/chatwoot/chatwoot/pull/9793/files#diff-e9818a3dc710645a0b30499e346262312210d60d68b54679e252769b012007b1R518

It was a bug that showed up after moving from bus to mitt.
